app/models/generic_object.rb

```diff
@@ -0,0 +1,103 @@
+class GenericObject < ApplicationRecord
+  validates :name, :presence => true, :uniqueness => true
+
+  belongs_to :generic_object_definition
+  has_many   :custom_attributes, :as => :resource, :dependent => :destroy, :autosave => true
+
+  TYPE_MAP = {
+    :boolean     => ActiveRecord::Type::Boolean.new,
+    :datetime    => ActiveRecord::Type::DateTime.new,
+    :time        => ActiveRecord::Type::Time.new,
+    :float       => ActiveRecord::Type::Float.new,
+    :integer     => ActiveRecord::Type::Integer.new,
+    :string      => ActiveRecord::Type::String.new,
+  }
+
+  def initialize(klass, attributes = {})
+    definition_object(klass)
+    raise "Generic Object Definition is required for creating a Generic Object." unless @definition_object
+    attributes[:generic_object_definition] = @definition_object
+    super(attributes)
+  end
+
+  def definition_object(klass = nil)
+    @definition_object ||= klass ? GenericObjectDefinition.find_by(:name => klass) : generic_object_definition
+  end
```

Comment:
If the .new is modified like I suggest, then this method just goes away I think.